.github/workflows/autogpt-server-cd.yml

```diff
@@ -0,0 +1,60 @@
+name: AutoGPT Server Docker Build and Deploy
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - 'rnd/autogpt_server/**'
+
+env:
+  PROJECT_ID: agpt-dev
+  IMAGE_NAME: agpt-server-dev
+  REGION: us-central1
+  GKE_CLUSTER: dev-gke-cluster
+  GKE_ZONE: us-central1-a
+  DEPLOYMENT_NAME: autogpt-server
+  HELM_CHART_PATH: rnd/infra/helm/autogpt-server
+
+jobs:
+  build-push-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0.2.1
+        with:
+          project_id: ${{ env.PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+
+      - name: Configure Docker
+        run: gcloud auth configure-docker ${{ env.REGION }}-docker.pkg.dev
+
+      - name: Build Docker image
+        working-directory: rnd/autogpt_server
+        run: docker build -t ${{ env.REGION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.IMAGE_NAME }}:${{ github.sha }} .
```

Comment:
Are we able to add a step to populate the cache before building so we can reuse costly layers - such as linux update layer and poetry install layer - we would also need to update our docker file structure to support this.